File: LeetCode/Java Solutions/Same Tree/same.java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        // If both trees are null, they are the same
        if (p == null && q == null) {
            return true;
        }
        // If one of the trees is null, they are not the same
        if (p == null || q == null) {
            return false;
        }
        // If the current nodes' values are different, the trees are not the same
        if (p.val != q.val) {
            return false;
        }
        // Recursively compare the left and right subtrees
        return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
    }
}

File: LeetCode/Python Solutions/Same Tree/same.py
from typing import Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        # If both nodes are None, they are the same (end of branch)
        if not p and not q:
            return True
        # If one node is None and the other is not, trees are not the same
        if not p or not q:
            return False
        # If the values of current nodes do not match, trees are not the same
        if p.val != q.val:
            return False
        # Recursively check the left and right subtrees
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
